src/main/cpp/tiv.cpp

```diff
@@ -382,6 +382,26 @@ void emit_image(const cimg_library::CImg<unsigned char> & image, int flags) {
 }
 
 
+struct size {
+  size(unsigned int in_width, unsigned int in_height) :
+	width(in_width), height(in_height) {
+  }
+  size(cimg_library::CImg<unsigned int> img) :
+	width(img.width()), height(img.height()) {
+  }
+  unsigned int width;
+  unsigned int height;
+};
+size operator*(size lhs, double scale) {
+  return size(lhs.width*scale, lhs.height*scale);
+}
+
+size fit_within(size container, size object) {
```

Comment:
Why not make this a method of size (returning a new instance, preserving the original)

Reply:
I couldn't find a name and API that clearly expressed which was the container and which the contained. This was my best idea (which may not be saying much). I found `maxThumbSize.fit_within(size(original))` or flipped around unclear.

Reply:
You could argue the same for the parameter order O:) As said in the comment above, a certain style simplifies maintenance for me. That's not saying one or the other option is "better".